### DISCUSSION TITLE
Pushing added files to a second branch in a workflow

### DISCUSSION THREAD

**Discussion by raihaan123** [community/community#24951] [Actions]
<p>Hi,</p>
<p>For reference, this is the repo I’m working on: <a href="https://github.com/raihaan123/ValiRocket" rel="noopener nofollow ugc">https://github.com/raihaan123/ValiRocket</a></p>
<p>My workflow involves a lengthy python script which executes under certain conditions, and at the end I create a new file within python. My question is, where does this file get saved if I don’t state any file path, since if I try to run the usual commands (within Python using os.system()) to git push to the second branch, nothing happens after the initial commit (i.e. it doesn’t see anything to add and commit)?</p>
<p>If I want to save the created file within a directory in the repo I am currently in, what file directory do I use (in Python running on Ubuntu)? I know my commands to push to the second branch work (as it pushed it successfully when it didn’t exist so there was a clear delta).</p>
<p>Many thanks!</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="raihaan123" data-post="1" data-topic="165034">
<div class="title">
<div class="quote-controls"></div>
 raihaan123:</div>
<blockquote>
<p>My workflow involves a lengthy python script which executes under certain conditions, and at the end I create a new file within python. My question is, where does this file get saved if I don’t state any file path,</p>
</blockquote>
</aside>
<p>What do you mean by writing a file without any path? That is technically possible for unnamed temporary files, but I assume if you want to commit the file that’s not what you’re trying to do.</p>

**Comment by raihaan123** [comment]
<p>Thanks for the reply,</p>
<p>By no path, I just mean this:</p>
<pre><code class="lang-auto">    with open("parsed.rocket", "wb") as f:
        pickle.dump(full_root, f)
</code></pre>
<p>which is basically me saving a binary file to the current working directory.  When run on my computer, it works fine and places parsed.rocket into the root of the repository (since that’s my working directory), but I’m assuming that’s not the case for GitHub Actions? Because it doesn’t seem to save where I expect…</p>
<p>Am I missing something obvious?</p>
<p>Many thanks!</p>

**Comment by airtower-luna** [comment]
<p>Ah, I see. The common term is a relative path. The default working directory for Actions is set in the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context"><code>github.workspace</code> variable</a> (or <a href="https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables"><code>GITHUB_WORKSPACE</code> environment variable</a>), and that’s also where your repository ends up if you use <code>actions/checkout</code> without options that say otherwise.</p>
<p>The title of your post mentions a second repository, do you have a more complex structure than checking out into the default workspace?</p>

**Comment by raihaan123** [comment]
<p>My apologies, I meant ‘second branch’, I’ve corrected it now.</p>
<p>I am not using <code>actions/checkout</code> but rather <code>os.system("git checkout -b ...")</code> from inside my python script - do I still need to include this in the yml action?</p>
<p>(As you may have gathered I am relatively new to GitHub actions)</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Running Git commands directly should work as well, <code>actions/checkout</code> also calls Git commands. :wink:</p>
<p>With <code>os.system()</code> the working directory of the Git command will be the same as that of your Python script. If the file is written without errors but not where you expect it, maybe add a step that searches for it? Something like this:</p>
<pre><code class="lang-yaml">- run: find . -name parsed.rocket
</code></pre>
<p>Or leave out everything after the dot to just list <em>every</em> file in the workspace. :grin:</p>

**Comment by raihaan123** [comment]
<p>Thanks so much!! I was able to isolate the issue(s), one was that I needed to lead file names with a ./ to get it onto the right directory - found this using your trick - and the other was some clumsy git from my side…</p>
